mono/mini/interp/interp.c

```diff
@@ -3881,8 +3881,24 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 				gpointer unboxed = mono_object_unbox_internal (this_arg);
 				sp [0].data.p = unboxed;
 			}
-retry_callvirt_fast:
-			if (imethod->code_type == IMETHOD_CODE_INTERP) {
+
+			InterpMethodCodeType code_type = imethod->code_type;
+
+			g_assert (G_UNLIKELY (code_type == IMETHOD_CODE_UNKNOWN) ||
+				code_type == IMETHOD_CODE_INTERP || code_type == IMETHOD_CODE_COMPILED);
+
+			if (G_UNLIKELY (code_type == IMETHOD_CODE_UNKNOWN)) {
+				MonoMethodSignature *sig = mono_method_signature_internal (imethod->method);
+				if (mono_interp_jit_call_supported (imethod->method, sig))
+					code_type = IMETHOD_CODE_COMPILED;
+				else
+					code_type = IMETHOD_CODE_INTERP;
+				imethod->code_type = code_type;
+			}
+
+			g_assert (code_type == IMETHOD_CODE_INTERP || code_type == IMETHOD_CODE_COMPILED);
+
```

Comment:
This is not needed.

Reply:
I removed the `g_assert`.
I guess, it is obvious enough, to human and compiler.